pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -59,6 +59,12 @@ const (
 // DefaultRouteName is the name assigned to a route generated by default in absence of a virtual service.
 const DefaultRouteName = "default"
 
+// maxRegExProgramSize defines the max regx complexity supported. 1024 is a safe default and should work
+// for most cases. We should look to make it configurable if this is not sufficient.
+// Note that this is different from length of regex.
+// Refer to https://github.com/google/re2/blob/a98fad02c421896bc75d97f49ccd245cdce7dd55/re2/re2.h#L287 for details.
+const maxRegExProgramSize = 1024
```

Comment:
@howardjohn r2e fuzzers actually validate till 9999 https://github.com/google/re2/blob/a98fad02c421896bc75d97f49ccd245cdce7dd55/re2/fuzzing/re2_fuzzer.cc#L50 - But I think we can start with 1024 to be safe and if any one complains - we can make it configurable/set it to 9999. WDYT?

Reply:
The non-re2 limit was 1024 though. So prior to 1.3 the hard max was 1024 as far as I know. And no one has complained about this, so I don't think we have any need to make it configurable. They can always do an EnvoyFilter if they really want to shoot themselves in the foot and make it so large.

Reply:
Oh program size is not length.. I didn't realize that.

Reply:
Yeah. It is not. But I guess 1024 should satisfy most cases - if it is not as I mentioned we can extend or they have an option of Envoy filter anyways as you mentioned.